src/dut.rs

```diff
@@ -307,13 +308,23 @@ impl DutInfo {
             }
         }
         if keys.contains(&"dut_id") {
-            let model = values.get("model");
-            let serial = values.get("serial");
-            if let (Some(Ok(model)), Some(Ok(serial))) = (model, serial) {
+            let serial: String = if let Some(Ok(serial)) = values.get("serial") {
+                serial.to_string()
+            } else {
+                // Some DUTs don't have serial number. So use MAC address,
+                let serial = if let Some(Ok(mac)) = values.get("mac") {
+                    mac.replace(':', "")
```

Comment:
how about adding some prefix to explicitly state "this is not a real serial number" to avoid confusion?
e.g. "NoSerial" so the "serial" will be "NoSerialAABBCCDDEEFF"

Reply:
Also, macaddr should be canonicalized to lowercase to avoid issues later (e.g. matched DUT not found since both upper / lower cases are used in different datasources)
c.f. https://www.ieee802.org/1/files/public/docs2020/yangsters-smansfield-mac-address-format-0420-v01.pdf

Reply:
Hmm, what about "MAC_AABBCCDDEEFF" for short?
Also, it already uses lowercase for "mac" value because `ip addr` shows it in lowercase.

Reply:
I want to avoid using "_" in the "serial" part of dut_id since sometimes model name includes "_" (e.g. sarien_signed) and it makes it difficult to separate the model part and serial part.

Also, there is no validation that mac is lowercase, so making sure they are lower case here is a good practice. (ideally, the length of the mac addr should be checked as well...)

And, it looks like there is a serial number that contains a string "mac" like "0MAC91ZH..." so just using MAC as a prefix may be confusing. I understand shorter is better, but with the command completion, it should be not a huge problem.

What do you think...?

Reply:
At first, I think canonicalizing the mac address is another issue, and should be done where the "mac" value is got.

And I wonder if we should be so cautious about using (temporary) mac addresses as serial numbers. What do you think a problematic scenario?

Reply:
> At first, I think canonicalizing the mac address is another issue, and should be done where the "mac" value is got.
yes, we should do the same thing on the logic that actually getting mac addr from the DUT. However, doing the same thing again here is not harmful, so I recommend doing that here. 

> And I wonder if we should be so cautious about using (temporary) mac addresses as serial numbers. What do you think a problematic scenario?

First, we want to make sure the "temporary" DUT_ID does not collide with the normal, valid "DUT_ID", since it can cause an inaccessible DUT when using a DUT_ID for `lium dut shell` or something. This is why we want to have a distinguishable prefix for such a pseudo-serial number.  Second, without output validation and if for some reason the command returns mac addr based DUT_ID in uppercase / lowercase randomly, you can't ssh sometimes because of the DUT_ID mismatch. For example, 
```
lium dut shell --dut `lium dut info --dut ${IP} | jq -r .dut_id` -- echo "OK"
```
should always succeed for all the valid `${IP}`, but if the DUT_ID returned by `lium dut info` is not stable, it cannot be accomplished.

Reply:
Ok, so you might concern that the existing serial number is similar (or same as) the mac address. Let me adding `NoSerial`.